app/src/components/ActionsPanel.js

```diff
@@ -1,137 +1,157 @@
-import React from 'react'
-import { Button, GU, Info, Field, SidePanel, TextInput } from '@aragon/ui'
+import React, { useCallback, useMemo, useState } from 'react'
+import {
+  Button,
+  GU,
+  Info,
+  Field,
+  SidePanel,
+  TextInput,
+  useSidePanelFocusOnReady,
+  useTheme,
+} from '@aragon/ui'
+import wrap from '../assets/wrap.svg'
+import { fromDecimals, toDecimals } from '../utils'
 
-const initialState = {
-  amount: '',
-}
+// Any more and the number input field starts to put numbers in scientific notation
+const MAX_INPUT_DECIMAL_BASE = 6
 
 const WrapTokensPanel = React.memo(
-  ({
-    panelState,
-    onAction,
-    action,
-    info,
-    erc20TokenSymbol,
-    wrappedTokenSymbol,
-  }) => {
+  ({ action, info, onAction, outsideToken, panelState, wrappedToken }) => {
     return (
       <SidePanel
         title={action + ' tokens'}
         opened={panelState.visible}
         onClose={panelState.requestClose}
-        onTransitionEnd={panelState.onTransitionEnd}
       >
         <WrapTokensPanelContent
-          onAction={onAction}
           action={action}
           info={info}
-          erc20TokenSymbol={erc20TokenSymbol}
-          wrappedTokenSymbol={wrappedTokenSymbol}
-          panelOpened={panelState.didOpen}
+          onAction={onAction}
+          outsideToken={outsideToken}
+          wrappedToken={wrappedToken}
         />
       </SidePanel>
     )
   }
 )
 
-class WrapTokensPanelContent extends React.PureComponent {
-  static defaultProps = {
-    onAction: () => {},
-  }
-  state = {
-    ...initialState,
-  }
-  componentWillReceiveProps({ panelOpened }) {
```

Comment:
Note that this manual focus handle and state resetting was made obsolete with `@aragon/ui@1.0.0`'s SidePanel and the exposed `useSidePanelFocusOnReady` hook.